flink-streaming-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSink.java

```diff
@@ -35,24 +36,28 @@
 	private static final Logger LOG = LoggerFactory.getLogger(RMQSink.class);
 
 	private String QUEUE_NAME;
-	private String HOST_NAME;
+	private RMQConnectionConfig rmqConnectionConfig;
 	private transient ConnectionFactory factory;
 	private transient Connection connection;
 	private transient Channel channel;
 	private SerializationSchema<IN> schema;
 
-	public RMQSink(String HOST_NAME, String QUEUE_NAME, SerializationSchema<IN> schema) {
-		this.HOST_NAME = HOST_NAME;
+	/**
+	 * @param rmqConnectionConfig The RabbiMQ connection configuration {@link RMQConnectionConfig}.
+	 * @param QUEUE_NAME The queue to publish messages to.
+	 * @param schema A {@link SerializationSchema} for turning the Java objects received into bytes
+     */
+	public RMQSink(RMQConnectionConfig rmqConnectionConfig, String QUEUE_NAME, SerializationSchema<IN> schema) {
+		this.rmqConnectionConfig = rmqConnectionConfig;
 		this.QUEUE_NAME = QUEUE_NAME;
 		this.schema = schema;
 	}
 
 	/**
 	 * Initializes the connection to RMQ.
 	 */
-	public void initializeConnection() {
-		factory = new ConnectionFactory();
-		factory.setHost(HOST_NAME);
+	public void initializeConnection() throws Exception {
+		factory = rmqConnectionConfig.getConnectionFactory();
```

Comment:
Another comment which is something that has been like this before: Could you remove the initializeConnection() method and move all its content into the open() method.
That reduces unnecessary boilerplate. 

Thank you ;)